express/dataset.py

```diff
@@ -0,0 +1,212 @@
+from abc import abstractmethod
+import argparse
+import json
+from pathlib import Path
+
+import dask.dataframe as dd
+
+from express.component_spec import ExpressComponent, kubeflow2python_type
+from express.manifest import Manifest, Subset, Index
+
+
+class FondantDataset:
+    """Wrapper around the manifest to download and upload data into a specific framework.
+
+    Uses Dask Dataframes for the moment.
+    """
+
+    def __init__(self, manifest):
+        self.manifest = manifest
+
+    @property
+    def project_name(self):
```

Comment:
not all clouds have the concept of `project_name`. What's the reason for adding this to the manifest?

Reply:
fsspec requires the full path, like "gcs://soy-audio-379412_datasets/pokemon.parquet". This includes the project name, bucket name and blob path.

Reply:
Can probably be addressed in a follow-up PR, would focus on getting the pipeline to run on GCP first